### ISSUE TITLE
Compute Sale total price in backend

### ISSUE THREAD

**Issue by cghislai** [Valuya/comptoir#25]
Sale and SaleItem prices should ba computed in backend
- Fetch item current Price when crating an ItemSale
- compute itemSale prices on create/update
- compute Sale prices on create/update

**Comment by ymajoros** [comment]
ItemSale is created on front side when an item is added to a sale, and is then POSTed. Sale will be adjusted by backend and it should be fetched by front to update displayed price.

If a discount is applied, ItemSale is adapted, PUT, sale calculated by backend. Frontend fetches it, asynchronously, and total price is shown.

OK for you?

**Comment by cghislai** [comment]
Ok, I discovered in the mean time the calc* method in sale service.
I just pushed a pull request (#26) to get it working with the latest frontend commits.

So i PUT an itemsale with saleRef, itemRef, quantity. The backend saves it in database, I fetch it back to get the item vatExclusive amount. I then fetch back the sale to get the updated vat and vatExclusive total amount.

 You seem to suggest I compute the itemSale price in the frontend, adapting it whith the possible discount. What about global sale discounts?
